stdlib/public/core/StringCore.swift

```diff
@@ -12,7 +12,7 @@
 
 /// The core implementation of a highly-optimizable String that
 /// can store both ASCII and UTF-16, and can wrap native Swift
-/// _StringBuffer or NSString instances.
+/// _StringBuffer or immutable literals.
```

Comment:
does that account for NSConstantString or NSTaggedPointerString as "immutable literals"?